mapviz_plugins/src/odometry_plugin.cpp

```diff
@@ -278,25 +278,70 @@ namespace mapviz_plugins
       PrintInfo("OK");
     }
   }
+  
+
+  void OdometryPlugin::Paint(QPainter* painter, double x, double y, double scale)
+  {
+    int interval = ui_.show_timestamps->value();
+    if (interval == 0)
+      return;
+
+    QTransform tf = painter->worldTransform();
+    QFont font("Helvetica", 10);
+    painter->setFont(font);
+    painter->save();
+    painter->resetTransform();
+
+    //set the draw color for the text to be the same as the rest
+    QPen pen(QBrush(ui_.color->color()), 1);
+    painter->setPen(pen);
+
+    std::list<StampedPoint>::iterator it = points_.begin();
+    int i = 0;//used to alternate between rendering text on some points
+    for (; it != points_.end(); ++it)
+    {
+      if (it->transformed && i++ % interval == 0)
```

Comment:
Side-effect increment/decrement operators are dangerous for a number of reasons; in this case, because their behavior is too easy to parse incorrectly. Presumably, you intend the postfix behavior here, but there is no good reason to be unclear. In general, prefer code clarity over brevity.

Move the increment outside of the conditional.